### PR TITLE
Fix TLS protocols used in tests

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Fixes https://github.com/aspnet/AspNetCore-Internal/issues/2490

Disabling tests failing in https://github.com/aspnet/AspNetCore-Internal/issues/2490.

### CODE REVIEW

**Comment by Tratcher** [comment]
This is actually a bit concerning, these tests aren't doing anything that should be OS specific.

The only difference I see is that they specify SslProtocols.None rather than SslProtocols.Tls11 | SslProtocols.Tls12.

Win7 defaults topped out at TLS 1.0. https://support.microsoft.com/en-us/help/3154520/support-for-tls-system-default-versions-included-in-the-net-framework. Kestrel defaults to 1.1 & 1.2 so this may explain the conflict.
https://github.com/aspnet/AspNetCore/blob/c2e2d7d1359ef8caf5460562adfe46b7f4f68df0/src/Servers/Kestrel/Core/src/HttpsConnectionAdapterOptions.cs#L27

Try changing the test to SslProtocols.Tls11 | SslProtocols.Tls12 instead.

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsTests.cs [hunk 1]
```diff
@@ -380,7 +380,7 @@ public async Task ClientAttemptingToUseUnsupportedProtocolIsLoggedAsDebug()
             Assert.Equal(LogLevel.Debug, loggerProvider.FilterLogger.LastLogLevel);
         }
 
-        [Fact]
+        [ConditionalFact]
```
Comment (BrennanConroy) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsTests.cs]: Change back to `Fact` and fix those compile errors

Comment (JunTaoLuo) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsTests.cs]: Gah. Should have ran it locally.

**Review by halter73 — APPROVED** [review]
LGTM. This matches the default value for SslProtocols in HttpsConnectionAdapterOptions, so it makes sense.

**Comment by anurse** [comment]
I updated the PR to close https://github.com/aspnet/AspNetCore-Internal/issues/2490 when merged since we believe it will solve the issue.

**Comment by JunTaoLuo** [comment]
Looks like the test is fixed! Thanks @Tratcher.

**Comment by JunTaoLuo** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=199321

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2570

**Pull request merged**